BootloaderCommonPkg/Include/Library/CryptoLib.h

```diff
@@ -77,6 +93,7 @@ RsaVerify (
   CONST UINT8              *Signature,
   CONST UINT32              SignatureLen,
   CONST UINT8               SignatureType,
+  CONST UINT8               HashType,
```

Comment:
Two comments:
1) Do we need 'SignatureType' argument? It does NOT convey any information.

2) I believe this method is to verify RSA signature using PKCS1v15 scheme. Should we reflect as such in the API name itself (RsaVerify_PKCS1v15)? In future, we may need to support PSS scheme to and then we can have a separate method (i.e. RsaVerify_PSS)?

Reply:
"SignatureType" is changed to sign_type. Comments were to be updated.

Yes, we could change the API name as RsaVerify_PKCS1v15 to reflect PKCS1v15.

Reply:
The first argument 'PubKey' should completely define what kind of PubKey it is : i.e. 2K or  3K.
HashType will define what kind of hashing algo was used during signature generation.

What purpose does 'sign_type' argument serve here?

Reply:
At higher API layer, we should use generic function name such as SignatureVerify () so that it will work for whatever new algorithm we want to use in future .   Algorithm should be a parameter.  Internally inside library, we can have more specific function implementation.

Reply:
"sign_type"/"sign_len" -  this info should be already available in calling interfaces.
Reusing the info, instead of calculations again.